src/_node/html/apis.ts

```diff
@@ -1,611 +1,154 @@
-  subNodeUids.map((subNodeUid) => {
-    const subNode = tree[subNodeUid];
-    const nodeData = subNode.data as THtmlNodeData;
-    if (nodeData.isFormatText) {
-      const text = nodeData.data;
-      const textParts = text.split(getLineBreaker(osType));
-      const singleLine = textParts.length === 1;
-      const lastPart = textParts.pop();
-      const newLastPart = " ".repeat(lastPart?.length || 0 + indentSize);
-      nodeData.data =
-        textParts.join(getLineBreaker(osType)) +
-        (singleLine ? "" : getLineBreaker(osType)) +
-        newLastPart;
-    }
-  });
-};
-let htmlContent = "";
-let htmlContentInApp = "";
+      isEntity: true,
+      children: [],
 
-export const parseHtml = (
-  content: string,
-  keepNodeUids: null | boolean = false,
-  nodeMaxUid: TNodeUid = "",
-): THtmlParserResponse => {
-  let _nodeMaxUid = keepNodeUids === false ? 0 : Number(nodeMaxUid);
-  // parse the html content
+      data: {
+        childNodes: htmlDom.childNodes,
 
-  const dom = parse5.parse(content, {
-    sourceCodeLocationInfo: true,
-    scriptingEnabled: true,
-    onParseError: (err) => {},
-  });
+        valid: true,
 
-  let appDom: Document;
+        nodeName: "",
+        tagName: "",
+        textContent: "",
 
-  htmlContent = content;
-  // parse the html content
-  const tmpTree: TNodeTreeData = {};
-  function preprocessNodes(dom: Document) {
-    appDom = dom;
+        attribs: {},
 
-    // build root node
-    tmpTree[RootNodeUid] = {
-      uid: RootNodeUid,
-      parentUid: null,
-      name: RootNodeUid,
-      isEntity: false,
-      children: [],
-      data: { valid: true },
-      sourceCodeLocation: {
-        startLine: 0,
-        startCol: 0,
-        startOffset: 0,
-        endLine: 0,
-        endCol: 0,
-        endOffset: 0,
+        codeRange: {
+          startLine: 0,
+          startCol: 0,
+          startOffset: 0,
+          endLine: 0,
+          endCol: 0,
+          endOffset: 0,
+        },
       },
     };
-
-    // build depth-1 seed nodes
-    const seedNodes: TNode[] = [];
-    appDom.childNodes.forEach((node) => {
-      const uid = String(++_nodeMaxUid) as TNodeUid;
+    const seedNodes: THtmlNode[] = [nodeTree[RootNodeUid]];
+    let _uid = 0;
+
+    const getHtmlNodeAttribs = (
+      uid: TNodeUid,
+      attrs: { name: string; value: string }[],
+    ): THtmlNodeAttributes => {
+      const attribs: THtmlNodeAttributes = {
+        [HtmlNodeUidAttribNameInApp]: uid,
+      };
+      attrs.map((attr) => {
+        attribs[attr.name] = attr.value;
+      });
+      return attribs;
+    };
+    const proceedWithNode = (
+      uid: TNodeUid,
+      parentUid: TNodeUid,
+      node: THtmlDomNode,
+      nodeTree: THtmlNodeTreeData,
+    ) => {
       const {
         startLine = 0,
         startCol = 0,
         startOffset = 0,
         endLine = 0,
         endCol = 0,
         endOffset = 0,
-      } = node.sourceCodeLocation || {};
+      } = node.data.codeRange || {};
```

Comment:
let's keep it `sourceCodeLocation` only as we get the same from parser.

Reply:
sure